# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/rustc/hotfix-llvm-17.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

warning: llvm-wrapper/PassWrapper.cpp:673:46: error: no matching
function for call to 'llvm::PGOOptions::PGOOptions(const char*&, const
char [1], const char [1],
llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem>&,
llvm::PGOOptions::PGOAction, llvm::PGOOptions::CSPGOAction, bool&)'

From 71958da4854176c50a8b12470b956d5c7ed11817 Mon Sep 17 00:00:00 2001
From: Krasimir Georgiev <krasimir@google.com>
Date: Wed, 12 Jul 2023 09:30:31 +0000
Subject: [PATCH] llvm-wrapper: adapt for LLVM API change

Adapts the wrapper for LLVM commit
https://github.com/llvm/llvm-project/commit/546ec641b4b1bbbf9e66a53983b635fe85d365e6.

Found by the experimental rust + LLVM @ HEAD bot: https://buildkite.com/llvm-project/rust-llvm-integrate-prototype/builds/20723#01894922-ed5d-4830-81f6-a27fb82ec8c7/210-645
---
 compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp b/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp
index c43a02724773..eb3d67e720f2 100644
--- a/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp
+++ b/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp
@@ -667,6 +667,7 @@ LLVMRustOptimize(
     assert(!PGOUsePath && !PGOSampleUsePath);
     PGOOpt = PGOOptions(PGOGenPath, "", "",
 #if LLVM_VERSION_GE(17, 0)
+                        "",
                         FS,
 #endif
                         PGOOptions::IRInstr, PGOOptions::NoCSAction,
@@ -675,6 +676,7 @@ LLVMRustOptimize(
     assert(!PGOSampleUsePath);
     PGOOpt = PGOOptions(PGOUsePath, "", "",
 #if LLVM_VERSION_GE(17, 0)
+                        "",
                         FS,
 #endif
                         PGOOptions::IRUse, PGOOptions::NoCSAction,
@@ -682,6 +684,7 @@ LLVMRustOptimize(
   } else if (PGOSampleUsePath) {
     PGOOpt = PGOOptions(PGOSampleUsePath, "", "",
 #if LLVM_VERSION_GE(17, 0)
+                        "",
                         FS,
 #endif
                         PGOOptions::SampleUse, PGOOptions::NoCSAction,
@@ -689,6 +692,7 @@ LLVMRustOptimize(
   } else if (DebugInfoForProfiling) {
     PGOOpt = PGOOptions("", "", "",
 #if LLVM_VERSION_GE(17, 0)
+                        "",
                         FS,
 #endif
                         PGOOptions::NoAction, PGOOptions::NoCSAction,

warning: llvm-wrapper/SymbolWrapper.cpp: In function 'void* LLVMRustGetSymbols(char*, size_t, void*, LLVMRustGet
SymbolsCallback, LLVMRustGetSymbolsErrorCallback)':
warning: llvm-wrapper/SymbolWrapper.cpp:42:3: error: 'SmallString' was not declared in this scope
warning: llvm-wrapper/SymbolWrapper.cpp:42:18: error: 'SymNameBuf' was not declared in this scope
warning: llvm-wrapper/SymbolWrapper.cpp:60:22: error: 'ErrorBuf' was not declared in this scope
warning: llvm-wrapper/SymbolWrapper.cpp:70:22: error: 'ErrorBuf' was not declared in this scope
warning: llvm-wrapper/SymbolWrapper.cpp:83:22: error: 'ErrorBuf' was not declared in this scope
From 6ddf9128b2b55f9def80af57f7353d2521527c6a Mon Sep 17 00:00:00 2001
From: Krasimir Georgiev <krasimir@google.com>
Date: Fri, 14 Jul 2023 12:10:29 +0000
Subject: [PATCH] llvm-wrapper: update for LLVM API change

No functional changes intended.

Adds an include for llvm::SmallString. Previously, this must have been
implicitly provided by some of the existing headers. With recent LLVM
changes, not anymore:
https://buildkite.com/llvm-project/rust-llvm-integrate-prototype/builds/20776#01895448-44a4-4a1e-8407-9d41d0186132/209-690
---
 compiler/rustc_llvm/llvm-wrapper/SymbolWrapper.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/compiler/rustc_llvm/llvm-wrapper/SymbolWrapper.cpp b/compiler/rustc_llvm/llvm-wrapper/SymbolWrapper.cpp
index 0493d6b05d03..bf00d11edf6d 100644
--- a/compiler/rustc_llvm/llvm-wrapper/SymbolWrapper.cpp
+++ b/compiler/rustc_llvm/llvm-wrapper/SymbolWrapper.cpp
@@ -7,6 +7,7 @@
 // * https://github.com/llvm/llvm-project/blob/8ef3e895ad8ab1724e2b87cabad1dacdc7a397a3/llvm/include/llvm/Object/ArchiveWriter.h
 // * https://github.com/llvm/llvm-project/blob/8ef3e895ad8ab1724e2b87cabad1dacdc7a397a3/llvm/lib/Object/ArchiveWriter.cpp
 
+#include "llvm/ADT/SmallString.h"
 #include "llvm/IR/LLVMContext.h"
 #include "llvm/Object/ObjectFile.h"
 
